components/rhs_comment/rhs_comment.jsx

```diff
@@ -324,13 +324,13 @@ export default class RhsComment extends React.Component {
         }
 
         let options;
-        if (isEphemeral) {
+        if (isEphemeral && (this.state.hover || this.state.dropdownOpened || this.state.showEmojiPicker)) {
```

Comment:
I think the change here is not needed. `createRemovePostButton` is not expensive as such. We can skip `this.state.dropdownOpened || this.state.showEmojiPicker` though even if we consider skipping initial render

Reply:
Most of the perf was actually because of emoji picker i think

Reply:
The point of havving the `dropdownOpened` and `showEmojiPicker` there is to not "close" the menu when you remove the hovering if the dropdown or the emoji picker is opened.

Reply:
Ohh, sorry, now I understand what you mean. yes, you are right.

Reply:
Do Ephermal posts have emojipicker or dropdowns?

Reply:
My bad, I misunderstood your comment. now I changed it. PTAL.

Reply:
On an unrelated note: This pattern has a flaw wrt to accessibility as we cannot navigate through this actions using keyboard.

Reply:
Yes, but is the same pattern used in the center channel. I think we have to redesign how this works to make it more performant and accessible.